peer/brontide.go

```diff
@@ -703,6 +703,23 @@ func (p *Brontide) Start() error {
 
 	p.startTime = time.Now()
 
+	// Before launching the writeHandler goroutine, we send any channel
```

Comment:
If we abstract out `loadActiveChannels`, or split it into: load chans and load messages, then we can write a unit test here to try to trigger the panic. IIUC, it relies on concurrent access to the brontide state machine, so the two goroutines need to line up directly. This may be easier to trigger with the `-race` flag on.

Reply:
I have a unit test that detects the race.  Will send a PR once I clean it up.

Reply:
https://github.com/lightningnetwork/lnd/pull/8198